### DISCUSSION TITLE
Re: Resizing Screen Query....

### DISCUSSION THREAD

**Discussion by snaidamast** [MonoGame/MonoGame#8584] [Graphics]
I have finally gotten my first _Military Simulation Workbench_ hexagon map working properly.  

All the hexagons appear to be placed in their correct positions as the screen snapshot shows below.  My question is, when I manually resize the map (which I have implemented the code for) and I resize the screen's borders over part of the hexagon map, the resized screen's borders erase that part part of the map the borders have gone over.  After I resize the screen's borders back to their original positions, which then will again show the entire map-board, the hexagon map is repainted fully but there is a delay in this happening.  

Should I refresh the viewable display back-buffer with the original RenderTarget2D during the resizing process then to avoid such a re-painting delay???

Thank you...

![SampleHexagonMap_8x6](https://github.com/user-attachments/assets/ba2a3d79-6c0e-409a-b329-154776b3ad39)

**Comment by KelsamGames** [comment]
I believe I've seen what you're describing before. Do you perhaps have something in your update/draw loop that stop the game from running when out of fucus? If so, I'd try disabling that just to see if that's what's happening. If not, it is possible that MG just doesn't run during window resizing. I'd say try the WindowsDX vs DesktopGL MG targets and see if it changes.

**Reply by snaidamast** [reply]
Thank you for your notes...

There is nothing in my game loop that cause it to stop when I ma resizing the window.  But it does appear that the screen can't repaint itself until the resizing to the original screen size is stopped.  

I am considering testing re-displaying the hidden buffer where the map was originally drawn during a re-sizing process.

**Reply by snaidamast** [reply]
Thank you for your notes...

There is nothing in my game loop that cause it to stop when I ma resizing the window.  But it does appear that the screen can't repaint itself until the resizing to the original screen size is stopped.  

I am considering testing re-displaying the hidden buffer where the map was originally drawn during a re-sizing process.

**Comment by MonogameGuy** [comment]
The game does not generally repaint while the window is being resized. 

To minimize the impact, use a fixed border (non-resizable) window and resize via menu option.

**Reply by KelsamGames** [reply]
It's also worth checking how it behaves in a GL build vs DX, lower-level windowing stuff like this can differ a lot based on what is being used to render it.

**Comment by arevcukigor** [comment]


**Reply by MonogameGuy** [reply]
@arevcukigor   What does a russian troll site have to do with this post?

**Reply by KelsamGames** [reply]
@MonogameGuy it's a spambot
